### PR TITLE
Filter logs files by `flowlog id` in AWS Wodle

### LINKED ISSUE #18901: AWS wodle shows abnormal behavior with the number of generated alerts
|Wazuh version|Component|
|---|---|
| 4.6.0 | wazuh-aws-wodle |

## Description

During the analysis of the issue https://github.com/wazuh/wazuh/issues/18843 an error was found when the Wazuh manager master nodes started showing errors due to lack of space, with up to 16GiB of alerts

```
[root@wazuh-manager-master-0 wazuh-user]# du -sh /var/ossec/logs/alerts
16G /var/ossec/logs/alerts
```

Performing an analysis, @vcerenu has detected that too many alerts like the following are being registered:

```
{"timestamp":"2023-09-08T17:43:00.769+0000","rule":{"level":3,"description":"AWS VPC Flow: [ACCEPT] - Interface: eni-xxx - Protocol : 6.","id":"80401","firedtimes":1078390,"mail":false,"groups":["amazon","aws","aws_vpcflow"]},"agent":{" id":"000","name":"wazuh-manager-master-0"},"manager":{"name":"wazuh-manager-master-0"},"id":"1694194980.9868307968", "cluster":{"name":"wazuh1","node":"master"},"decoder":{"name":"json"},"data":{"integration":"aws"," aws":{"log_info":{"log_file":"vpc/AWSLogs/xxxx/vpcflowlogs/us-west-1/2020/12/22/166157441623_vpcflowlogs_us-west-1_fl-04d16c61b3339cdf6_202031222T 35Z_7b747144.log.gz","s3bucket" :"wazuh-aws-wodle"},"version":"2","account_id":"xxxx","interface_id":"eni-xxxx","srcaddr":"10.0.1.34","dstaddr": "xxxx","srcport":"22","dstport":"47916","protocol":"6","packets":"18","bytes":"5241","start": "2020-12-22T13:35:47Z","end":"2020-12-22T13:36:47Z","action":"ACCEPT","log_status":"OK","source":"vpc ","aws_account_id":"xxxx"}},"location":"Wazuh-AWS"}
```

This is confirmed by looking at the alerts in the WUI as there are `13,371,724` alerts from `wazuh-aws-wodle` out of a total of `13,402,731`. These are too many alerts for an environment of a day and a half, plus @teddytpc1 has commented that this does not happen in the 4.5.2 demo environment

It is necessary to investigate why so many alerts are being generated in such a short period of time for said integration.

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|---|
| #18901 |

## Description

This PR closes #18901. Fixes a problem of iterating multiple times over the same log files when there is more than one `FlowLog` configured. 

Nows iterates correctly over the bucket files

```
root@wazuh-master:/var/ossec# wodles/aws/aws-s3 -t vpcflow -b wazuh-aws-wodle -l vpc -p prod -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG: +++ Working on XXXXXXXXXXXX - us-east-1
DEBUG: +++ Working on XXXXXXXXXXXX - us-east-2
DEBUG: +++ Working on XXXXXXXXXXXX - us-west-1
DEBUG: +++ Marker: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-02814797492******_20230914T1220Z_8f603f75.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-02814797492******_20230914T1225Z_0bfbc9f8.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-02814797492******_20230914T1230Z_d6f741b6.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-02814797492******_20230914T1235Z_0765e09f.log.gz
DEBUG: +++ DB Maintenance
DEBUG: +++ Marker: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1220Z_116b8bf0.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1220Z_6fa110c9.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1225Z_0616aa79.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1225Z_284986f8.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1230Z_9cadc649.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1230Z_f41ba6de.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1235Z_358fcd1b.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1235Z_bf0a7776.log.gz
DEBUG: ++ Found new log: vpc/AWSLogs/XXXXXXXXXXXX/vpcflowlogs/us-west-1/2023/09/14/XXXXXXXXXXXX_vpcflowlogs_us-west-1_fl-0bb6085e007******_20230914T1240Z_13a45908.log.gz
DEBUG: +++ DB Maintenance
```

## Integration Tests

```python
root@ip-172-31-82-0:/home/ubuntu/wazuh-qa/tests/integration/test_aws# pytest --tb=line --tier 0 -x --disable-warnings
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/ubuntu/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.4, html-3.1.1, testinfra-5.0.0
collected 198 items / 16 deselected / 182 selected

test_basic.py ................                                                                                                                                                                                [  8%]
test_discard_regex.py .................                                                                                                                                                                       [ 18%]
test_log_groups.py ..                                                                                                                                                                                         [ 19%]
test_only_logs_after.py ...............................                                                                                                                                                       [ 36%]
test_parser.py ..........................                                                                                                                                                                     [ 50%]
test_path.py ..........................................                                                                                                                                                       [ 73%]
test_path_suffix.py .........                                                                                                                                                                                 [ 78%]
test_regions.py ........................                                                                                                                                                                      [ 91%]
test_remove_from_bucket.py ...sss.........                                                                                                                                                                    [100%]

====================================================================== 179 passed, 3 skipped, 16 deselected, 6 warnings in 3704.14s (1:01:44) =======================================================================
root@ip-172-31-82-0:/home/ubuntu/wazuh-qa/tests/integration/test_aws# pytest --tb=line --tier 1 -x --disable-warnings
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/ubuntu/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.4, html-3.1.1, testinfra-5.0.0
collected 198 items / 182 deselected / 16 selected

test_only_logs_after.py ..............x.                                                                                                                                                                      [100%]

======================================================================= 15 passed, 182 deselected, 1 xfailed, 6 warnings in 368.84s (0:06:08) =======================================================================
```

### CODE REVIEW

File: wodles/aws/aws_s3.py [hunk 1]
```diff
@@ -1206,6 +1206,17 @@ def _print_no_logs_to_process_message(self, bucket, aws_account_id, aws_region,
 
         debug(self.empty_bucket_message_template.format(**message_args), 1)
 
+    def _filter_bucket_files(self, bucket_files, **kwargs):
```
Comment (GGP1) [inline wodles/aws/aws_s3.py:1209]: Add docstring and `bucket_files` type hint

File: wodles/aws/aws_s3.py [hunk 2]
```diff
@@ -1831,9 +1835,11 @@ def db_maintenance(self, aws_account_id=None, aws_region=None, flow_log_id=None)
         except Exception as e:
             print(f"ERROR: Failed to execute DB cleanup - AWS Account ID: {aws_account_id}  Region: {aws_region}: {e}")
 
-    def get_vpc_prefix(self, aws_account_id, aws_region, date, flow_log_id):
-        return self.get_full_prefix(aws_account_id, aws_region) + date \
-            + '/' + aws_account_id + '_vpcflowlogs_' + aws_region + '_' + flow_log_id
+    def _filter_bucket_files(self, bucket_files, **kwargs):
```
Comment (GGP1) [inline wodles/aws/aws_s3.py:1838]: Add docstring and `bucket_files` type hint

**Review by GGP1 — CHANGES_REQUESTED** [review]
Re-run failed [check](https://github.com/wazuh/wazuh/pull/19002/checks?check_run_id=16762368105)

**Review by GGP1 — APPROVED** [review]
LGTM!

**Review by Selutario — APPROVED** [review]

**Pull request merged**